subprojects/logging/src/test/groovy/org/gradle/internal/logging/sink/GroupingProgressLogEventGeneratorTest.groovy

```diff
@@ -20,17 +20,31 @@ import org.gradle.api.logging.LogLevel
 import org.gradle.internal.logging.OutputSpecification
```

Comment:
I used `Semaphore` to synchronized and stage the output in the functional test. Could you give me a bit of feedback?